js-miniapp-sdk/src/miniapp.ts

```diff
@@ -35,6 +35,16 @@ interface MiniAppFeatures {
    * @returns The Promise of share info action state from injected side.
    */
   shareInfo(info: ShareInfoType): Promise<string>;
+
+  /**
+   * @returns Username saved in the host app user profile
+   */
+  getUserName(): Promise<string>;
```

Comment:
We're starting to have a lot of methods in `MiniApp` (and will have a lot more to come). So I was thinking, maybe we should start name-spacing them or somehow providing a logical separation between related functionality. This would allow us to separate them in the source code and also provide a more clear interface to the user (hopefully).

We've kind of started to do that with the `Ad` and `Platform` interfaces, but I think we could take this even further by moving each interface into its own separate implementation which is then set as a property of `MiniApp`. So basically I mean something like this:

```
interface UserInfoProvider {
  getUserName(): Promise<string>;
}

/** @internal */
class UserInfo implements UserInfoProvider {
  getUserName(): Promise<string> {
    return this.bridge.getUserName();
  }
}

export class MiniApp {
  user: UserInfoProvider = new UserInfo();
}
```

Or another solution is to completely separate `UserInfoProvider` into its own class which must be imported and used on its own, but that doesn't really fit with what we've done so far of having everything under `MiniApp`.

What do you think?

Reply:
Honestly I wanted to separate `UserInfoProvider` into its own class, import and work. Not just `UserInfoProvider` but also other interfaces.  I thought I will do refactor in another ticket